### PR TITLE
Reservecommand

### LINKED ISSUE #70: Create test for ReserveCommand + Parser
Please reread ReserveCommand class, if there is any problem, tell me asap.

### PR DESCRIPTION
**Pull request by sancharib-git**
***do not merge yet***

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ReserveCommand.java [hunk 1]
```diff
@@ -67,15 +66,15 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException("No room with this number");
         }
         model.bookRoom(new Booking(payee.get(), room.get(), reserveDuration));
-        return new CommandResult(generateSuccessMessage(roomNum, payee.get()));
+        return new CommandResult(generateSuccessMessage(payee.get()));
 
     }
 
     /**
      * Generates a command execution success message based on whether the remark is added to or removed from
```
Comment (tanphatls987) [inline src/main/java/seedu/address/logic/commands/ReserveCommand.java]: Rewrite this comment

File: src/test/java/seedu/address/logic/commands/ReserveCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    private static final Person ALICE = new PersonBuilder().build();
+    private static final Room DEFAULT_ROOM = new RoomBuilder().build();
+    private static final LocalDateTime FROM_DATE = LocalDateTime.parse("2020-12-12");
+    private static final LocalDateTime TO_DATE = LocalDateTime.parse("2020-12-23");
+    private static final Reserve RESERVE_STUB = new Reserve(ALICE, DEFAULT_ROOM, FROM_DATE, TO_DATE);
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void executeReserve_success() {
```
Comment (tanphatls987) [inline src/test/java/seedu/address/logic/commands/ReserveCommandTest.java]: Look good

File: src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        String userInput = PREFIX_ID + ID.toString()
+                + PREFIX_ROOMNUMBER + ROOM.getRoomNum()
+                + PREFIX_FROMDATE + FROM_DATE
+                + PREFIX_TODATE + TO_DATE;
+        ReserveCommand expectedCommand = new ReserveCommand(ID, ROOM.getRoomNum(), FROM_DATE, TO_DATE);
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_missingCompulsoryField_failure() {
```
Comment (tanphatls987) [inline src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java:43]: See format for reserveCommand, I guess this is just copy and haven't done yet?

File: src/test/java/seedu/address/testutil/RoomBuilder.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import seedu.address.model.hotel.Tier;
+import seedu.address.model.hotel.person.Person;
+import seedu.address.model.ids.RoomId;
+
+
+/***
+ * A utility class to help build rooms
+ */
+public class RoomBuilder {
+    public static final Person ALICE = TypicalPersons.ALICE;
```
Comment (tanphatls987) [inline src/test/java/seedu/address/testutil/RoomBuilder.java]: no need for ALICE to be here

File: src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java (hunk 3 above)
Comment (sancharib-git) [reply src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java:43]: I don't understand what you're saying, it's not copied. The test follows the ReserveCommand format i.e. tests for missing parameters of ReserveCommand.

Comment (tanphatls987) [reply src/test/java/seedu/address/logic/parser/ReserveCommandParserTest.java:43]: sorry it is indeed correct

**Comment by sancharib-git** [comment]
able to merge, will add in remaining tests and bookings into database by next week

**Review by youaremysky99 — APPROVED** [review]

**Pull request merged**